Networking/Sources/Networking/Networking.swift

```diff
@@ -1,6 +1,36 @@
+import Alamofire
+import Foundation
+
 public struct Networking {
-    public private(set) var text = "Hello, World!"
+    static let shared = Networking()
 
     public init() {
     }
+    
+    public func fetchAllProducts<T: Codable>(expectingReturnType: T.Type, completion: @escaping ((Result<T?,Error>) -> Void)){
+        guard let url = URLs.shared.getProductsURL() else { return }
+        
+        let method = Alamofire.HTTPMethod(rawValue: HTTPMethod.get.rawValue)
+        let headers = Alamofire.HTTPHeaders([:])
+        
+        AF.request(url, method: method, parameters: nil, encoding: JSONEncoding.default, headers: headers).response { response in
+            switch response.result {
+            case .failure(_):
+                completion(.failure(GeneralErrors.callingAPI))
+                
+            case .success(_):
+                guard let data = response.data else {return}
+                do{
+                    let jsonData = try JSONDecoder().decode(T.self, from: data)
+                    completion(.success(jsonData))
+                    print(jsonData)
+                }catch{
+                    completion(.failure(GeneralErrors.failedData))
+                    print("Error")
+                }
+                
+            }
+        }
```

Comment:
This code will be duplicated in each request, any idea how to make this layer more generic?